From 1ec2999f9206bd0d81c30b60fda45528e002c58e Mon Sep 17 00:00:00 2001
From: Wencai You <wencai.you@amlogic.com>
Date: Fri, 12 Jul 2019 21:06:41 +0800
Subject: [PATCH] audio: add pass through fixed volume control [1/1]

PD#SWPL-10451

Problem:
    in pass through ,press volue control
    volume controller is seen

Solution:
    fixed pass through property

 Verify:
        u212

Change-Id: I07d6608dd1ce15a1ecccd35f20617cf5a4176bf4
Signed-off-by: Wencai You <wencai.you@amlogic.com>
---
 services/audioflinger/Threads.cpp | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/services/audioflinger/Threads.cpp b/services/audioflinger/Threads.cpp
index 9bc2d58..b61b5d1 100644
--- a/services/audioflinger/Threads.cpp
+++ b/services/audioflinger/Threads.cpp
@@ -545,7 +545,7 @@ void AudioFlinger::ThreadBase::exit()
 {
     ALOGV("ThreadBase::exit");
 
-    property_set("audio.passthrough.enable", "false");
+    //property_set("audio.passthrough.enable", "false");
 
     // do any cleanup required for exit to succeed
     preExit();
@@ -1886,7 +1886,22 @@ void AudioFlinger::PlaybackThread::preExit()
     // FIXME this is using hard-coded strings but in the future, this functionality will be
     //       converted to use audio HAL extensions required to support tunneling
     status_t result = mOutput->stream->setParameters(String8("exiting=1"));
-    ALOGE_IF(result != OK, "Error when setting parameters on exit: %d", result);
+    audio_format_t          mHALFormat;
+    status_t result2 =mOutput->stream->getFormat(&mHALFormat);
+     switch (mHALFormat) {
+         case AUDIO_FORMAT_E_AC3:
+         case AUDIO_FORMAT_AC3:
+         case AUDIO_FORMAT_DTS:
+         case AUDIO_FORMAT_DTS_HD:
+             property_set("audio.passthrough.enable", "false");
+             break;
+         default:
+             break;
+     }
+
+    ALOGE_IF(result != OK, "result2 = %d Error when setting parameters on exit: %d",result2,result);
+
+
 }
 
 // PlaybackThread::createTrack_l() must be called with AudioFlinger::mLock held
@@ -2527,7 +2542,7 @@ void AudioFlinger::PlaybackThread::readOutputParameters_l()
             property_set("audio.passthrough.enable", "true");
             break;
         default:
-            property_set("audio.passthrough.enable", "false");
+        //    property_set("audio.passthrough.enable", "false");
             break;
     }
     LOG_ALWAYS_FATAL_IF(result != OK, "Error when retrieving output stream format: %d", result);
-- 
1.9.1

